openlcsd/flow/tasks.py

```diff
@@ -944,27 +944,48 @@ def get_module_components_from_corgi(context, engine):
     engine.logger.info("Start to get module components data...")
     mc = ParentComponentsAsync(
         config.get('CORGI_API_PROD'), context.get('package_nvr'))
-    context['components'] = mc.get_components_data('RHEL_MODULE')
+    context['components'] = mc.get_components_data('RPMMOD')
     engine.logger.info("Finished getting module components data.")
 
 
+def get_module_srpm(context, engine):
+    """
+    The get_module_components_from_corgi gets the RPM components that includes
+    the rpms and srpms. This function gets the srpm builds for rhel module.
+    """
+    components = context.get('components')
+    rpm_components = components.get('RPM')
+    engine.logger.info("Start to get srpm builds of the module...")
+    srpms = []
+    for rpm in rpm_components:
+        if rpm.get('arch') == 'src':
+            nvr = [rpm.get('name'), '-', rpm.get('version'), '-',
+                   rpm.get('release')]
+            srpms.append(''.join(nvr))
+    engine.logger.info("Finish to get srpm builds of the module.")
+    context['nvr_list'] = srpms
+
+
 def fork_detail_type_components_imports(
         context, engine, nvr_list, src_dir, comp_type):
     """
     In a source container, it has different type of component. The different
-    components have different src_dir. This function will be defined the post
-    data for the different type component.
+    components have different src dir. This function will be defined the post
+    data for the source container srpm components' tasks. It could also be
+    same with rhel module srpm components except the src dir.
```

Comment:
I cannot understand what is `except the src dir`

Reply:
The src dir is the source directory. If the components are the container child components, they have been downloaded when the source image had been downloaded. The src dir should be defined in container import. If the components are the module child components, they are the general srpms downloading that the src dir is different for the forked task.